.circleci/config.yml

```diff
@@ -0,0 +1,52 @@
+version: 2.1
+
+commands:
+  after_failure:
+    parameters:
+      when:
+        type: string
+    steps:
+      - run:
+          name: On failure action
+          command: |
+            cat /var/log/nuodb/nuoadmin.log*
+            grep ssl /opt/nuodb/etc/nuoadmin.conf
+          when: <<parameters.when>>
+      
+
+jobs:
+  build_n_run:
+    docker:
+      - image: nuodb/nuodb-ce:latest
+        user: root
+    resource_class: small
+    steps:
+      - checkout
+      - run:
+          command: dnf install make -y
+          name: Install make
+      - run:
+          name : Install python dependencies
+          command: make install
+      - run:
+          name : Configure and run NuoDB Admin
+          command: |
+            sudo -u nuodb "/opt/nuodb/etc/nuoadmin" tls $NUO_SET_TLS
+            sudo -u nuodb "/opt/nuodb/etc/nuoadmin" start
+      - run:
+          name: Run test
+          command: make test
+      - store_artifacts:
```

Comment:
This saves the _codecov_ report only which is useful. I think that in addition to this, you should have two more steps:
* save the _pytest_ JUnit report using `store_test_results` step. Once this is done, you should be able to see the tests in "Tests" tab of the job in CircleCI app.
* save the NuoAdmin logs in case of a failure using `store_artifacts` step.